pkg/services/pluginsintegration/plugininstaller/service.go

```diff
@@ -2,31 +2,76 @@ package plugininstaller
 
 import (
 	"context"
+	"errors"
+	"runtime"
+	"strings"
 
 	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/plugins"
 	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginstore"
 	"github.com/grafana/grafana/pkg/setting"
 )
 
 type Service struct {
-	features featuremgmt.FeatureToggles
-	log      log.Logger
+	cfg             *setting.Cfg
+	features        featuremgmt.FeatureToggles
+	log             log.Logger
+	pluginInstaller plugins.Installer
+	pluginStore     pluginstore.Store
 }
 
-func ProvideService(cfg *setting.Cfg, features featuremgmt.FeatureToggles) *Service {
+func ProvideService(cfg *setting.Cfg, features featuremgmt.FeatureToggles, pluginStore pluginstore.Store, pluginInstaller plugins.Installer) *Service {
 	s := &Service{
-		features: features,
-		log:      log.New("plugin.installer"),
+		features:        features,
+		log:             log.New("plugin.backgroundinstaller"),
+		cfg:             cfg,
+		pluginInstaller: pluginInstaller,
+		pluginStore:     pluginStore,
 	}
 	return s
 }
 
 // IsDisabled disables background installation of plugins.
 func (s *Service) IsDisabled() bool {
-	return !s.features.IsEnabled(context.Background(), featuremgmt.FlagBackgroundPluginInstaller)
+	return !s.features.IsEnabled(context.Background(), featuremgmt.FlagBackgroundPluginInstaller) ||
+		len(s.cfg.InstallPlugins) == 0
 }
 
 func (s *Service) Run(ctx context.Context) error {
```

Comment:
Nit: It might be nice to wrap this with some instrumentation (duration of install, how many, etc.). Logs are probably fine for now, but metrics could be useful to alert on down the road

Reply:
note that the goal of this feature is mostly for on prem at the moment so we don't need instrumentation. If we decide we want to use it in cloud at some point we can add it.

Reply:
I was thinking more in terms of helping us diagnose issues (for example high latency) in case of an escalation